### PR TITLE
Can now re-number or re-name blocks

### LINKED ISSUE #4699: MeshModifier idea
Would a MeshModifier that changes one blockID to another blockID be of interest and appropriate for inclusion in the framework?

I use VTK to build my meshes and annoyingly it doesn't have sidesets.  So i create a heap of blocks and then use SideSetsFromNormals or SideSetsBetweenBlocks.  But then i have a billion blocks (well, in reality, only thousands) that are otherwise useless.  I want to change them back to their "original" block numbers that they would have had, had I been able to use sidesets.

Comments?

### PR DESCRIPTION
**Pull request by WilkAndy**
Fixes #4699

Please help:  Functionally this is what i wanted to achieve.  BUT, i don't really like the error checking on lines 32-45 of RenameBlock.C.  I want the user to input exactly one of old_block_id or old_block_name, and my way of ensuring this seems somehow quite clunky.

Also, i'd appreciate someone checking the modify() function as i don't know what i'm doing - i just copied from other parts of MOOSE.

I probably went overboard with the error checking - i'll remove some of the exception tests if you like.

### CODE REVIEW

File: framework/src/meshmodifiers/RenameBlock.C [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    _old_block_id = _mesh_ptr->getSubdomainIDs(getParam<std::vector<SubdomainName> >("old_block_name"));
+
+
+  if (isParamValid("new_block_id"))
+  {
+    _new_block_id = getParam<std::vector<SubdomainID> >("new_block_id");
+    if (_new_block_id.size() != _old_block_id.size())
+      mooseError("RenameBlock: The vector of old_block information must have the same length as the vector of new_block information\n");
+    for (unsigned i = 0 ; i < _old_block_id.size() ; ++i)
+      for (MeshBase::element_iterator el = _mesh_ptr->getMesh().active_elements_begin(); el != _mesh_ptr->getMesh().active_elements_end(); ++el)
```
Comment (jwpeterson) [inline framework/src/meshmodifiers/RenameBlock.C]: Not a big deal, but you might reverse the order of these two loops so you only do one loop over elements.

**Comment by jwpeterson** [comment]
> I probably went overboard with the error checking - i'll remove some of the exception tests if you like.

I'm OK with the number of tests, but this might be a good place to use `cli_args` in the `tests` file to activate the different MeshModifier error conditions.  That way you'd have only one input file, and your tests file would control which one gets turned on, e.g.
``` 
cli_args = 'MeshModifiers/active=re_id1
...
cli_args = 'MeshModifiers/active=re_id2
```
There are differing opinions on this in general (you don't want to hide too much functionality in the tests file) but I think in this case it's warranted.

File: framework/src/meshmodifiers/RenameBlock.C (hunk 1 above)
Comment (WilkAndy) [inline framework/src/meshmodifiers/RenameBlock.C]: OK, thanks @jwpeterson ,

a

**Comment by WilkAndy** [comment]
Implemented @jwpeterson 's suggestions

**Comment by permcody** [comment]
:+1:

**Comment by WilkAndy** [comment]
Thanks @permcody ,

a

**Pull request merged**